kernelci/api/helper.py

```diff
@@ -279,10 +279,17 @@ def create_job_node(self, job_config, input_node,
             'path': input_node['path'] + [job_config.name],
             'group': job_config.name,
             'artifacts': {},
+            'jobfilter': input_node['jobfilter'],
```

Comment:
You should use a variable, like `jobfilter = input_node.get('jobfilter')`: this way we ensure `jobfilter` (the variable) always exists, but is `None` if `input_node` doesn't have a `jobfilter` key. Otherwise this line (and the `if` below) would raise a `KeyError` exception., causing a crash.

Reply:
Updated